spec/tasks/notify_rake_spec.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+RSpec.describe "Notify rake tasks" do
+  describe "notify:send_email" do
+    let(:email_address) { "x@y.com" }
+    before { Rake::Task["notify:send_email"].reenable }
+
+    it "sends an email notification via GOV.UK Notify" do
+      template_id = SecureRandom.uuid
+      body = "Body content"
+      subject = "A subject line"
+
+      client = instance_double("Notifications::Client")
+      allow(Notifications::Client).to receive(:new).and_return(client)
+
+      expect(client).to receive(:send_email)
+        .with(email_address: email_address,
+              template_id: template_id,
+              personalisation: { body: body, subject: subject })
+
+      ClimateControl.modify(TEMPLATE_ID: template_id, BODY: body, SUBJECT: subject) do
+        Rake::Task["notify:send_email"].invoke(email_address)
+      end
```

Comment:
If we're able to go with my suggestion of using actionmailer then I think we could benefit from using action mailer in the test too. 

From [googling around](https://www.lucascaton.com.br/2010/10/25/how-to-test-mailers-in-rails-with-rspec/#) it looks like there's an array set on ActionMail::Base.deliveries that can be checked for what has been mailed. That array [seems to contain Mail objects](https://github.com/mikel/mail/blob/master/lib/mail/network/delivery_methods/test_mailer.rb#L41) so could asset that it appended to that array a mail object with the parameters you'd expect.

I think doing that would abstract a couple of the tests here from being very similar to the notify one and also be an interesting thing to learn about how we should test emails our sent functionally.